utils/python/CIME/case.py

```diff
@@ -277,6 +277,8 @@ def set_value(self, item, value, subgroup=None, ignore_type=False):
         then that value will be set in the file object and the file
         name is returned
         """
+        if item == "CASEROOT":
```

Comment:
We now have a _caseroot variable in the case object - make sure that it is updated if set_value("CASEROOT" is called.